drivers: tdk: Define the trigger as a pointer to allow dereferencing within the trigger handler callback.

janchri opened the pull request:
This is a proposal to fix the issue with CONTAINER_OF dereferencing in the trigger handler (see the following issue: https://github.com/zephyrproject-rtos/zephyr/issues/97124).

With this fix, it is possible to define a custom-made 

```
struct sensor_device {
...
struct sensor_trigger trigger;
...
}; 
struct sensor_device sdevice;
```

and by setting `sensor_trigger_set(dev, &sdevice.trigger, handler)` while initialisation, access in `handler() `via `CONTAINER_OF `to the underlying `struct sdevice`  is granted.

fixes: https://github.com/zephyrproject-rtos/zephyr/issues/97124

ubieda reviewed (dismissed):
Thanks for the fix!

kartben commented:
@janchri can you please update the commit messages to indicate it `Fixes zephyrproject-rtos/zephyr#97124`?

kartben commented:
@janchri can you please update the commit message to indicate it `Fixes zephyrproject-rtos/zephyr#97124`?

janchri committed:
drivers: sensor: tdk: redefine trigger as ptr to make use of CONTAINER_OF

Allows CONTAINER_OF inside trigger handler to be used to obtain a valid
ptr to the composite struct consisting of a struct sensor_trigger trig.

Fixes zephyrproject-rtos/zephyr#97124

Signed-off-by: Christoph Jans <jans.christoph@gmail.com>

ubieda reviewed (commented):
Should the commit title be one line or is it okay splitting it as it is in this PR?
